src/OS8104A.ts

```diff
@@ -0,0 +1,697 @@
+import { Gpio } from "onoff"
+import spi, { type SpiDevice, type SpiOptions } from "spi-device"
+import EventEmitter from "events"
+import { getRegisterConfig } from "./RegisterConfig"
+import { Registers } from "./Registers.js"
+const TRANSFER_SPEED = 180000
+
+export interface MostMessage {
+    targetAddressHigh: number
+    targetAddressLow: number
+    fBlockID: number
+    instanceID: number
+    fktId: number
+    opType: number
+    data: number[]
+}
+
+export interface Config {
+    freq: number
+    nodeAddressLow: number
+    nodeAddressHigh: number
+    groupAddress: number
+}
+
+export interface Stream {
+    sourceAddrHigh: number
+    sourceAddrLow: number
+    fBlockID: number
+    instanceID: number
+    sinkNr: number
+}
+
+export interface AllocResult {
+    loc1: number
+    loc2: number
+    loc3: number
+    loc4: number
+    cl: number
+    answer1: string
+    freeChannels: number
+}
+
+export interface SourceResult {
+    nodePos: number
+    group: number
+    logicalHigh: number
+    logicalLow: number
+}
+
+const options: SpiOptions = {
+    chipSelectHigh: false,
+    bitsPerWord: 8,
+    lsbFirst: false
+}
+
+export class OS8104A extends EventEmitter {
+    readonly freq: number
+    readonly spi: SpiDevice
+    readonly interrupt: Gpio
+    readonly fault: Gpio
+    readonly status: Gpio
+    getRegisterConfig: typeof getRegisterConfig
+    mostStatus: Gpio
+    reset: Gpio
+    nodeAddressBuf: Buffer
+    groupAddressBuf: Buffer
+    awaitAlloc: boolean
+    allocResult: AllocResult
+    awaitGetSource: boolean
+    getSourceResult: SourceResult | null
+    allocTimeout: ReturnType<typeof setTimeout> | null
+    streamAllocTimeout: ReturnType<typeof setTimeout> | null
+    delayTimer: ReturnType<typeof setTimeout> | null
+    getSourceTimeout: ReturnType<typeof setTimeout> | null
+    allocCheck: ReturnType<typeof setInterval> | null
+    lockInterval: ReturnType<typeof setInterval> | null
```

Comment:
This does not feel right, and then also leads to a bunch of null checks later when the requirement to clear an interval happens, looks like a `clearInterval` will accept a type of `number` so maybe thats the best route here

Reply:
You could use ` lockInterval?: NodeJS.Timer`

This means it can be undefined or a timer (the result of setInterval)
You don't need to initialize it as null - and clearInterval already accepts Timer or undefined

This can apply to all timers here (Timeouts are NodeJS.Timeout though)

Reply:
Worked perfectly

https://github.com/rhysmorgan134/SocketMost/blob/9ce8ed6645f6af46f293c8d8e9eb070ae297161b/src/driver/OS8104A.ts#L40-L45